src/Data/Group/Free.hs

```diff
@@ -72,6 +73,13 @@ instance Group (FreeGroup a) where
       . reverse
       . runFreeGroup
 
+instance Eq a => GroupOrder (FreeGroup a) where
+    -- TODO: It performs simplify each time @order@ is called.
+    --   Once "auto-simplify" is implemented, this
+    --   call of simplify should be removed.
+    order g | simplify g == mempty = Finite 1
+            | otherwise           = Infinite
```

Comment:
+1 We're looking at better ways to encode `FreeGroup` as suggested by others. Thanks for leaving the TODO.